selene-lib/src/lints/roblox_mismatched_udim2_new_arg_count.rs

```diff
@@ -0,0 +1,108 @@
+use super::*;
+use crate::ast_util::range;
+use std::convert::Infallible;
+
+use full_moon::{
+    ast::{self, Ast},
+    visitors::Visitor,
+};
+
+pub struct UDim2ArgCountLint;
+
+impl Lint for UDim2ArgCountLint {
+    type Config = ();
+    type Error = Infallible;
+
+    const SEVERITY: Severity = Severity::Error;
+    const LINT_TYPE: LintType = LintType::Correctness;
+
+    fn new(_: Self::Config) -> Result<Self, Self::Error> {
+        Ok(UDim2ArgCountLint)
+    }
+
+    fn pass(&self, ast: &Ast, context: &Context, _: &AstContext) -> Vec<Diagnostic> {
+        if !context.is_roblox() {
+            return Vec::new();
+        }
+
+        let mut visitor = UDim2CountVisitor::default();
+
+        visitor.visit_ast(ast);
+
+        visitor
+            .args
+            .iter()
+            .map(|mismatch| {
+                Diagnostic::new_complete(
+                    "roblox_mismatched_udim2_new_arg_count",
+                    format!(
+                        "UDim2.new takes 4 numbers, but {} were provided.",
+                        mismatch.num_provided
+                    )
+                    .to_owned(),
+                    Label::new(mismatch.call_range),
+                    vec![
+                        "help: did you mean to use UDim2.fromScale or UDim2.fromOffset instead?"
```

Comment:
If the parameters are between 0 and 1, suggest fromScale. Otherwise, suggest fromOffset.